public/utils/datasource-utils.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { DataSourceOption } from 'src/plugins/data_source_management/public/components/data_source_menu/types';
+
+export function createDataSourceQuery(dataSourceId: string) {
+  return { dataSourceId };
+}
+
+const DATASOURCEURLKEY = 'dataSource';
+
+export function getClusterInfoIfEnabled(dataSourceEnabled: boolean, cluster: DataSourceOption) {
+  if (dataSourceEnabled) {
+    return `for ${cluster.label || 'Local cluster'}`;
```

Comment:
Will cluster.label always be non-empty for a remote datasource? Is label a required field when configuring a datasource?

Reply:
Yes, my understanding is that label is always defined, and required during creation.